src/error/src/anyhow.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Define a newtype wrapper around [`anyhow::Error`].
+///
+/// # Motivation
+///
+/// [`anyhow::Error`] is good enough if one just wants to make the error type
+/// informative but not necessarily actionable. However, given a value of type
+/// [`anyhow::Error`], it is hard to tell which module or crate it comes from,
```

Comment:
I don't get how this macro solve this problem: `[`anyhow::Error`], it is hard to tell which module or crate it comes from`. It seems like defining a wrapper type of `anyhow::Error`.

Reply:
> It seems like defining a wrapper type of `anyhow::Error`.

Yes, it is. I'm talking about the type-erased error type, instead of the error source type (like `mysql::Error`).

By wrapping `anyhow::Error` into `ConnectorError`, upper crate can clearly find it come from the `connector` crate and not be confused with other error types. With bare `anyhow::Error`, the boundary will not be clear.